jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -123,7 +131,10 @@ object XGBoost extends Serializable {
           }
           val partitionItr = fromDenseToSparseLabeledPoints(trainingSamples, missing)
           val trainingSet = new DMatrix(new JDMatrix(partitionItr, cacheFileName))
-          booster = SXGBoost.train(trainingSet, xgBoostConfMap, round,
+          if (groupData != null) {
+            trainingSet.setGroup(groupData(partIndex).toArray)
+          }
```

Comment:
what's the difference to serialize groupData as a part of xgboostConfMap and an independent variable?